sys/cddl/contrib/opensolaris/uts/common/fs/zfs/zfs_znode.c

```diff
@@ -145,8 +145,9 @@ zfs_znode_cache_destructor(void *buf, void *arg)
 	znode_t *zp = buf;
 
 	ASSERT(!POINTER_IS_VALID(zp->z_zfsvfs));
-	ASSERT(ZTOV(zp) == NULL);
+	ASSERT(ZTOV(zp) != NULL);
```

Comment:
More than 2 years since I submitted this patch, so I'll need to look at the details. I think my patch is probably wrong because this assert would've been triggered otherwise. 

It's only strange to me that we call vn_free on a NULL vnode, look:
	ASSERT(ZTOV(zp) == NULL);
	vn_free(ZTOV(zp));

I'll take a look at this later.